packages/gatsby/cache-dir/loader.js

```diff
@@ -226,6 +230,54 @@ export class BaseLoader {
           // undefined if final result is an error
           return pageResources
         })
+
+        const staticQueryBatchPromise = Promise.all(
```

Comment:
Check for in flight requests and use cached results from before like we do for page data